modal/cli/environment.py

```diff
@@ -39,10 +40,25 @@ def __rich__(self):
 def list(json: Optional[bool] = False):
     envs = environments.list_environments()
     table_data = []
+
+    env_var_default = environ.get("MODAL_ENVIRONMENT") or None
+    cfg_default = str(config.get("environment")) or None
+    active_env_count = 0
     for item in envs:
-        is_active = item.name == config.get("environment")
-        row = [item.name, item.webhook_suffix, is_active if json else RenderableBool(is_active)]
+        is_active = item.name == cfg_default or item.default is True or item.name == env_var_default
+        if is_active:
+            active_env_count += 1
+        row = [item.name, item.webhook_suffix, cast(Union[Text, str], is_active if json else RenderableBool(is_active))]
         table_data.append(row)
+
+    if active_env_count > 1:
+        # set is_active column to false for all rows except for the env in config
+        active_env = env_var_default or cfg_default
+        for i in range(0, len(table_data)):
+            env = table_data[i]
```

Comment:
As Howard mentioned above I don't think you need to do this iteration but FYI in Python it's much cleaner to just have

```python
for env_row in table_data:
    ...
```